src/Containers/Timeline.tsx

```diff
@@ -0,0 +1,187 @@
+
+    const getTimelineItem = (Item: DataItem, index: number) => (
+        <TimelineItem key={`${index}_timeline`}>
+            <TimelineItemLeft color={separatorColor} isFirst={index === 0}>
+                <Item.Icon style={{ width: '15px', color: Item.color }} />
+            </TimelineItemLeft>
+            <TimelineItemRight color={textColor}>{Item.text}</TimelineItemRight>
+        </TimelineItem>
+    );
+    return (
+        <TimelineBox {...props}>
+            <Title color={titleColor}>
+                <strong>{title}</strong>
+            </Title>
+            <SubContentContainer>
+                <SubContent>
+                    <SmallTitle color={textColor}>
+                        {subContentLeftTitle}
+                    </SmallTitle>
+                    <BigContent color={figuresColor}>
+                        <strong>{subContentLeftAmountLocale}</strong>
+                    </BigContent>
+                </SubContent>
+                <SubContent>
+                    <SmallTitle color={textColor}>
+                        {subContentRightTitle}
+                    </SmallTitle>
+                    <BigContent color={figuresColor}>
+                        <strong>{subContentRightAmountLocale}</strong>
+                    </BigContent>
+                </SubContent>
+            </SubContentContainer>
+            <Separator width={separatorLength} color={separatorColor} />
+            {data.map((item, index) => getTimelineItem(item, index))}
+        </TimelineBox>
+    );
+};
+
+const TimelineItemRight = styled.div`
+    ${({ theme, color }): string => `
+font-size:${theme.font.size.small};
+color:${color};
+`}
+    margin-left:15px;
+`;
+
+interface TimelineItemLeftProps extends React.HTMLAttributes<HTMLDivElement> {
+    isFirst: boolean;
+}
+
+const TimelineItemLeft = styled.div<TimelineItemLeftProps>`
+    width: 40px;
+    height: 40px;
+    border-radius: 50%;
+    ${({ color }): string => `
+color:${color};
+`}
+    background-color:currentColor;
+    ${flex('center')}
+    ${({ isFirst }): string =>
+        isFirst
+            ? ''
+            : `
+    &::before {
+        content: '';
+        width: 0px;
+        height: 10px;
+        border: 2px solid currentColor;
+        position: relative;
+        top: -30px;
+        left: 7px;
+    }
+    `}
+`;
+
+const TimelineItem = styled.div`
+    ${flex('row', 'flex-start', 'center')}
+    margin-top:20px;
+`;
+
+interface SeparatorProps extends React.HTMLAttributes<HTMLDivElement> {
+    width: string | number;
+}
+
+const Separator = styled.div<SeparatorProps>`
+    ${({ width, color }): string => `
+width:${width}px;
+border:2px solid ${color};
+`}
+`;
+
+const BigContent = styled.div`
+    ${({ theme, color }): string => `
+font-size:${theme.font.size.h4};
+color:${color};
+`}
+`;
+
+const SmallTitle = styled.div`
+    ${({ theme, color }): string => `
+font-size:${theme.font.size.small};
+color:${color};
+`}
+    margin-bottom:5px;
+`;
```

Comment:
There are a few text options in the React-UI library for text, some of which use theme for color which will be beneficial